src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java

```diff
@@ -1,20 +1,33 @@
 package group1.mips_simulator.components.cpuParts;
 
+import group1.mips_simulator.Utility;
+import group1.mips_simulator.components.ROM;
+import group1.mips_simulator.components.instructionParts.Instruction;
+import group1.mips_simulator.components.memParts.Memory;
+
+import group1.mips_simulator.components.instructionParts.Instruction;
+import java.util.Vector;
+
 public class RomLoader {
+    Memory memory;
     /*
     read the boot program from the ROM and place it into memory in a location you designated.
     The ROM Loader then transfers control to the program which executes until completion or error
      */
-    String ROMLocation;
-
-    public RomLoader(){
-        this.ROMLocation = null;
+    public RomLoader(Memory mem){
+        memory = mem;
     }
-    public RomLoader(String directory){
-        this.ROMLocation = directory;
+
+    public void load(ROM rom){
+        Vector<Instruction> romData = rom.getInstructions();
+        for(int i = 0; i < romData.size(); i++){
+            writeToMemory((short) i, romData.get(i));
```

Comment:
Can we start this at an offset? Like at memory location 1024 or something? Mem location 0 is reserved.

Reply:
yep, on it